ethereum/contracts/ParachainClient.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.5;
+
+import "./BeefyClient.sol";
+import "./utils/MerkleProof.sol";
+import "./ScaleCodec.sol";
+import "./utils/MMRProofVerification.sol";
+
+contract ParachainClient {
+    using ScaleCodec for uint32;
+
+    BeefyClient public immutable beefyClient;
+    uint32 public immutable parachainID;
+    bytes4 public immutable encodedParachainID;
+
+    struct HeadProof {
+        uint256 pos;
+        uint256 width;
+        bytes32[] proof;
+    }
+
+    struct MMRLeafPartial {
+        uint8 version;
+        uint32 parentNumber;
+        bytes32 parentHash;
+        uint64 nextAuthoritySetID;
+        uint32 nextAuthoritySetLen;
+        bytes32 nextAuthoritySetRoot;
+    }
+
+    struct Proof {
+        bytes headPrefix;
+        bytes headSuffix;
+        HeadProof headProof;
+        MMRLeafPartial leafPartial;
+        MMRProof leafProof;
+    }
+
+    constructor(BeefyClient _client, uint32 _parachainID) {
+        beefyClient = _client;
+        parachainID = _parachainID;
+        encodedParachainID = ScaleCodec.encode32(_parachainID);
+    }
+
+    function verifyCommitment(bytes32 commitment, Proof calldata proof)
+        external
+        view
+        returns (bool)
```

Comment:
makes sense, just reminder to make sure we don't forget now that this no longer reverts on failure and that it is the responsibility of the consumer to check this.

maybe we should document this more strongly either in a comment or in README/solidity docs once we improve them